doc/hardware/porting/board_porting.rst

```diff
@@ -133,57 +171,127 @@ board directory to a separate repository once it's working.)
 
   The board directory name does not need to match the name of the board.
   Multiple boards can even defined be in one directory.
-  For example, for boards with multi-core SoC, a logical board might be created
-  for each core following the naming scheme `<board>_<soc-core>`, with definitions
-  for all of these different boards defined inside the same directory. This and
-  similar schemes are common for upstream vendor boards.
 
 Your board directory should look like this:
 
 .. code-block:: none
 
    boards/<ARCH>/plank
+   ├── board.yml
    ├── board.cmake
    ├── CMakeLists.txt
    ├── doc
    │   ├── plank.png
    │   └── index.rst
-   ├── Kconfig.board
+   ├── Kconfig.plank
    ├── Kconfig.defconfig
    ├── plank_defconfig
+   ├── plank_<identifier>_defconfig
    ├── plank.dts
+   ├── plank_<identifier>.dts
+   ├── plank_<identifier>.overlay
    └── plank.yaml
 
 Replace ``plank`` with your board's name, of course.
 
 The mandatory files are:
 
-#. :file:`plank.dts`: a hardware description in :ref:`devicetree
-   <dt-guide>` format. This declares your SoC, connectors, and any
-   other hardware components such as LEDs, buttons, sensors, or communication
-   peripherals (USB, BLE controller, etc).
+#. :file:`board.yml`: a YAML file describe the highlevel meta data of the board
+   such as the board name, SoCs, and variants.
+   Cpusets for multi-core SoCs are not described in this file as they are
+   inherited from the SoC's YAML description.
+
+#. :file:`plank.dts` or :file:`plank_<identifier>.dts`: a hardware description
+   in :ref:`devicetree <dt-guide>` format. This declares your SoC, connectors,
+   and any other hardware components such as LEDs, buttons, sensors, or
+   communication peripherals (USB, BLE controller, etc).
+
+#. :file:`Kconfig.plank`: the base software configuration for selecting SoC and
+   other board and SoC related setting. No Kconfig settings outside the board
+   and SoC tree must be selected. To select general Zephyr Kconfig settings the
+   :file:`Kconfig` file shall be used.
 
-#. :file:`Kconfig.board`, :file:`Kconfig.defconfig`, :file:`plank_defconfig`:
-   software configuration in :ref:`kconfig` formats. This provides default
-   settings for software features and peripheral drivers.
 
 The optional files are:
 
+- :file:`plank_<identifier>.overlay`: a hardware description overlay in
```

Comment:
I found this document, and tried creating a plank_<soc>_<variant>.overlay, but it did not work. This is because cmake/dts.cmake only has support for BOARD_REVISION here:

https://github.com/zephyrproject-rtos/zephyr/blob/3932ac8866f2c458a31030974993538d686f9667/cmake/modules/dts.cmake#L137

I wanted to see your intent on this. I'm also unsure if this impacts app overlays as I'm new.

Reply:
This is working, i.e. in `zephyr/samples/hello_world/build`:
```
mkdir ../boards
echo "blah" > ../boards/nrf5340dk_nrf5340_cpuapp_ns.overlay
cmake -GNinja -DBOARD=nrf5340dk//cpuapp/ns ..
...
-- Found BOARD.dts: /tmp/aa/zephyr/boards/nordic_nrf/nrf5340dk/nrf5340dk_nrf5340_cpuapp_ns.dts
-- Found devicetree overlay: /tmp/aa/zephyr/samples/hello_world/boards/nrf5340dk_nrf5340_cpuapp_ns.overlay
devicetree error: /tmp/aa/zephyr/samples/hello_world/boards/nrf5340dk_nrf5340_cpuapp_ns.overlay:1 (column 1): parse error: expected '/' or label reference (&foo)
```
Note: has to be in a `boards` directory due to a bug

Reply:
To be more precise, I was trying to add an overlay file to board/laird_connect/bl654_dvk called bl654_dvk_nrf52840_pa.overlay. The dts.cmake code ignoreed that. Is this a valid use of a .overlay file, or should they be limited to the samples/*/boards directory?

Reply:
> Note: has to be in a boards directory due to a bug

No, this was designed and documented to be that way.
Ref:
- https://github.com/zephyrproject-rtos/zephyr/pull/29990
- https://github.com/zephyrproject-rtos/zephyr/pull/31088

Reply:
> The dts.cmake code ignoreed that. Is this a valid use of a .overlay file, or should they be limited to the samples/*/boards directory?

when revision overlays was introduced in #29990 and #31088 it was purposely picked up only from `<appdir>/boards/` to not clutter more the `<appdir>` toplevel folder, but support for regular `<appdir>/<board>.overlay` was kept.

For HWMv2, then due to many boards will have the former `BOARD_NAME` identical to `BOARD_FULL_IDENTIFIER`, then we'll support this variant during initial transition, but long term, with all the possible combination of names, then I would advice to always place overlays under `<appdir>/boards/`.

Reply:
right. this part has been removed.